libqtile/backend/base.py

```diff
@@ -654,8 +654,19 @@ def finalize(self):
 
     def add_mirror(self, mirror: Drawer):
         """Keep details of other drawers that are mirroring this one."""
+        # If this is the first mirror for this object then we need to create a
+        # separate RecordingSurface for mirrors to access.
+        if not self.mirrors:
+            self._create_last_surface()
         self.mirrors.add(mirror)
 
+    def remove_mirror(self, mirror: Drawer):
+        try:
+            self.mirrors.remove(mirror)
+        except KeyError:
+            # This may get called multiple times but only the first call will work
+            pass
```

Comment:
From what I can tell, each mirror would be removed exactly once. In what situation can this method be called more than once for the same widget and get this key error? Same with `_Widget.remove_mirror`

Reply:
That was my expectation to but I hit the `KeyError` in practice.